src/server/worker/stats.go

```diff
@@ -0,0 +1,115 @@
+package worker
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	// PrometheusPort is the port the aggregated metrics are served on for scraping
+	PrometheusPort = int32(9090)
+
+	datumCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "pachyderm",
+			Subsystem: "user",
+			Name:      "datum_count",
+			Help:      "Number of datums counted by pipeline and state",
+		},
+		[]string{
+			"pipeline",
+			"job",
+			"state",
+		},
+	)
+
+	bucketFactor  = 2.0
+	bucketCount   = 20 // Which makes the max bucket 2^20 seconds or ~12 days in size
+	datumProcTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "pachyderm",
+			Subsystem: "user",
+			Name:      "datum_proc_time",
+			Help:      "Time processing user code",
```

Comment:
Another small nit, but I think this should be "Time running user code" or "Time processing datums." The first one seems a bit better to me.